conans/cli/commands/upload.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+import json
+import sys
+
+from conans.cli.command import conan_command, Extender, OnceArgument
+
+
+def output_upload_cli(info, out):
+    for uploaded_reference in info["uploaded_references"]:
+        out.write("{}".format(uploaded_reference))
+
+
+def output_upload_json(info, out):
+    myjson = json.dumps(info, indent=4)
+    out.write(myjson)
+
+
+@conan_command(group="Consumer", formatters={"cli": output_upload_cli,
+                                             "json": output_upload_json})
+def upload(*args, conan_api, parser, **kwargs):
+    """
+    Uploads a recipe and binary packages to a remote.
+
+    If no remote is specified, the first configured remote will be used.
+    """
+
+    parser.add_argument("pattern_or_reference", help="Pattern, recipe reference or package "
+                        "reference e.g., 'boost/*', 'MyPackage/1.2@user/channel', "
+                        "'MyPackage/1.2@user/channel:af7901d8bdfde621d086181aa1c495c25a17b137'",
+                        nargs="?")
+    parser.add_argument("-r", "--remote", action=OnceArgument,
+                        help="Upload to this specific remote")
+    parser.add_argument("--query", default=None, action=OnceArgument,
```

Comment:
This would be a perfect candidate to get out of here, and pipe the result of ``conan search --query`` into ``conan upload``